src/main/java/stsjorbsmod/cards/wanderer/ChainLightning.java

```diff
@@ -35,8 +37,42 @@ public ChainLightning() {
 
     @Override
     public void use(AbstractPlayer p, AbstractMonster m) {
-        ArrayList<AbstractMonster> targets = AbstractDungeon.getMonsters().monsters;
-        addToBot(new ChainLightningAction(p, m, targets, damage, magicNumber, AttackEffect.NONE));
+        ArrayList<AbstractMonster> targets = getRandomOrderMonsters(AbstractDungeon.getMonsters().monsters, m);
+        int[] damagesArray = getDamagesArray(targets);
+        addToBot(new ChainLightningAction(p, m, targets, damagesArray, magicNumber, AttackEffect.NONE));
+    }
+
+    private ArrayList<AbstractMonster> getRandomOrderMonsters(ArrayList<AbstractMonster> targets, AbstractMonster initialTarget) {
+        ArrayList<AbstractMonster> finalTargets = new ArrayList<>();
+        ArrayList<AbstractMonster> newTargets = new ArrayList<>();
+
+        newTargets.addAll(targets);
+        int nextTargetIndex = newTargets.indexOf(initialTarget);
+        while(true) {
+            finalTargets.add(newTargets.remove(nextTargetIndex));
+            if (newTargets.isEmpty()) break;
+            nextTargetIndex = AbstractDungeon.cardRandomRng.random(0, newTargets.size() - 1);
+        }
+        return finalTargets;
+    }
+
+    private int[] getDamagesArray(ArrayList<AbstractMonster> targets) {
+        int[] damagesArray = targets.stream().map(monster -> {
+            if (monster.halfDead || monster.isDying || monster.isEscaping) return 0;
+            int monsterDamage = getDamage(monster, damageMultiplier);
+            damageMultiplier += 1;
+            return monsterDamage;
+        }).mapToInt(i -> i).toArray();
+
+        damageMultiplier = 0;
+
+        return damagesArray;
+    }
+
+    private int getDamage(AbstractMonster monster, int multiplier) {
+        this.baseDamage = DAMAGE + (DAMAGE_PLUS_PER_HOP * multiplier);

```

Comment:
Setting `baseDamage` like this is going to have the side effect of ignoring Wrath bonuses. Instead of this line, you can add an override for `calculateBonusBaseDamage` based on `damageMultiplier` to get a similar effect that's more wrath-friendly, eg:

```java
@Override
protected int calculateBonusBaseDamage() {
    return magicNumber * damageMultiplier;
}
```